src/NativeScript/TNSRuntime.mm

```diff
@@ -36,6 +44,15 @@ - (instancetype)initWithApplicationPath:(NSString*)applicationPath {
         self->_applicationPath = [applicationPath copy];
         WTF::wtfThreadData().m_apiData = static_cast<void*>(self);
 
+#if PLATFORM(IOS)
+        VM* vm = self->_vm.get();
+        self->_memoryPressureNotificationSubscription = [[NSNotificationCenter defaultCenter] addObserverForName:UIApplicationDidReceiveMemoryWarningNotification object:nil queue:nil usingBlock:^(NSNotification *note) {
+            JSLockHolder lock(vm);
+            vm->heap.collect(FullCollection);
+            vm->heap.releaseDelayedReleasedObjects();
```

Comment:
I'm not sure this happens on every `unlock` - according to https://github.com/WebKit/webkit/blob/master/Source/JavaScriptCore/runtime/JSLock.cpp#L162 `JSLock::unlock` will call `Heap::releaseDelayedReleasedObjects` only when the lock recursion is zeroed out, and I'm not certain this condition would be met every time the memory pressure notification fires.

Even if it turns out to be redundant, at worst a second call to  `Heap::releaseDelayedReleasedObjects` would be a no-op, so I suggest leaving it as it is.